src/stream/src/executor/group_top_n.rs

```diff
@@ -196,33 +196,31 @@ where
             let pk_row = row_ref.row_by_indices(&self.internal_key_indices);
             let ordered_pk_row = OrderedRow::new(pk_row, &self.internal_key_order_types);
             let row = row_ref.to_owned_row();
+
             let mut group_key = Vec::with_capacity(self.group_by.len());
             for &col_id in &self.group_by {
                 group_key.push(row[col_id].clone());
             }
+            let pk_prefix = Row::new(group_key.clone());
 
             // If 'self.caches' does not already have a cache for the current group, create a new
             // cache for it and insert it into `self.caches`
-            let pk_prefix = Row::new(group_key.clone());
-            let entry = self.caches.entry(group_key);
-            match entry {
-                Occupied(_) => {}
-                Vacant(entry) => {
-                    let mut topn_cache = TopNCache::new(self.offset, self.limit, self.order_by_len);
-                    self.managed_state
-                        .init_topn_cache(Some(&pk_prefix), &mut topn_cache)
-                        .await?;
-                    entry.insert(topn_cache);
-                }
+            if let Vacant(entry) = self.caches.entry(group_key) {
+                let mut topn_cache = TopNCache::new(self.offset, self.limit, self.order_by_len);
+                self.managed_state
+                    .init_topn_cache(Some(&pk_prefix), &mut topn_cache)
+                    .await?;
+                entry.insert(topn_cache);
             }
+            let cache = self.caches.get_mut(&pk_prefix.0).unwrap();
 
             // apply the chunk to state table
             match op {
                 Op::Insert | Op::UpdateInsert => {
                     self.managed_state
                         .insert(ordered_pk_row.clone(), row.clone());
-                    self.caches.get_mut(&pk_prefix.0).unwrap().insert(
-                        ordered_pk_row,
+                    cache.insert(
+                        ordered_pk_row.skip(pk_prefix.size()),
```

Comment:
Can we directly do this when constructing the `ordered_pk_row` to avoid another allocation? 🥵

Reply:
~~The line above needs it.~~ It actually doesn't need it :think: